### PR TITLE
System Adding Student

### PR DESCRIPTION
**Pull request by katieann7**

### CODE REVIEW

File: student-information-management-application/src/main/java/com/student/information/management/add/app/model/student/Student.java [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+    }
+
+    public void setContactNumber(String contactNumber) {
+        this.contactNumber = contactNumber;
+    }
+
+    /**
+     * Validation for student id should be have a pattern like CT00-0000.
+     */
+    public static boolean isValidStudentId(String studentId) {
```
Comment (JustineDaveMagnaye) [inline student-information-management-application/src/main/java/com/student/information/management/add/app/model/student/Student.java:103]: Add validation if there are 2 same student Id

File: student-information-management-application/src/main/java/com/student/information/management/add/data/student/dao/impl/StudentDaoImpl.java [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+            statement.setString(10, student.getContactNumber());
+            int result = statement.executeUpdate();
+            return result == 1? true: false;
+
+        } catch (Exception e) {}
+        return false;
+    }
+
+    @Override
+    public Student addStudents(ResultSet rs) {
```
Comment (JustineDaveMagnaye) [inline student-information-management-application/src/main/java/com/student/information/management/add/data/student/dao/impl/StudentDaoImpl.java:74]: Remove this addStudents(ReseultSet rs) it is unused

File: student-information-management-application/src/main/java/com/student/information/management/add/data/student/dao/StudentDao.java [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+import java.sql.ResultSet;
+import java.util.List;
+/**
+ * This is the Student Dao Interface.
+ */
+public interface StudentDao {
+    List<Student> getAllStudents();
+    Student getStudentById(String student_id) throws RuntimeException;
+    boolean addStudent(Student student);
+    Student addStudents(ResultSet rs);
```
Comment (JustineDaveMagnaye) [inline student-information-management-application/src/main/java/com/student/information/management/add/data/student/dao/StudentDao.java:14]: Remove this addStudents(ReseultSet rs) it is unused

File: student-information-management-application/src/main/java/Main.java [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+        String email = sc.nextLine();
+        if (isValidEmail(email)) {
+            newStudent.setEmail(email);
+            System.out.println("Email set successfully.");
+        } else {
+            System.out.println("Invalid email format. Please enter a valid email.");
+            return;
+        }
+        System.out.print("Enter Address: ");
+        newStudent.setAddress(sc.nextLine());
```
Comment (JustineDaveMagnaye) [inline student-information-management-application/src/main/java/Main.java:88]: Where is the validation for String only for isValidAddress?

**Comment by katieann7** [comment]
All done. The no usage is still needed for program so I didn't remove it.

File: student-information-management-application/build.xml [hunk 5]
```diff
@@ -0,0 +1,17 @@
+<project name="student-information-management-application">
```
Comment (RoCS2024) [inline student-information-management-application/build.xml:1]: complete all build tasks

File: student-information-management-application/src/main/java/com/student/information/management/add/data/student/dao/StudentDao.java [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+import com.student.information.management.add.app.model.student.Student;
+
+import java.sql.ResultSet;
+import java.util.List;
+/**
+ * This is the Student Dao Interface.
+ */
+public interface StudentDao {
+    List<Student> getAllStudents();
+    Student getStudentById(String student_id) throws RuntimeException;
```
Comment (RoCS2024) [inline student-information-management-application/src/main/java/com/student/information/management/add/data/student/dao/StudentDao.java:12]: why is an interface throwing an exception?

File: student-information-management-application/src/main/java/com/student/information/management/add/data/student/dao/impl/StudentDaoImpl.java [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+import static com.student.information.management.add.data.utils.QueryConstant.*;
+/**
+ * This is the Student Dao Impl.
+ */
+public class StudentDaoImpl implements StudentDao {
+    Connection con = ConnectionHelper.getConnection();
+
+    @Override
+    public List<Student> getAllStudents() {
+        try {
```
Comment (RoCS2024) [inline student-information-management-application/src/main/java/com/student/information/management/add/data/student/dao/impl/StudentDaoImpl.java:23]: there is no return value when exception happens

File: student-information-management-application/src/main/java/com/student/information/management/add/data/student/dao/impl/StudentDaoImpl.java [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+            while(rs.next()) {
+                students.add(setStudent(rs));
+            }
+            return students;
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+    @Override
+    public Student getStudentById(String id) throws RuntimeException {
```
Comment (RoCS2024) [inline student-information-management-application/src/main/java/com/student/information/management/add/data/student/dao/impl/StudentDaoImpl.java:37]: fix the exception

File: student-information-management-application/src/main/java/com/student/information/management/add/app/model/student/Student.java [hunk 9]
```diff
@@ -0,0 +1,204 @@
...
+    private String religion;
+    private String email;
+    private String address;
+    private String contactNumber;
+
+    public Student(){
+
+    }
+
+    public Student(String studentId, String lastName, String firstName, String middleName, String sex, String birthday, String religion, String email, String address, String contactNumber) {
```
Comment (RoCS2024) [inline student-information-management-application/src/main/java/com/student/information/management/add/app/model/student/Student.java:27]: this constructor is useless

File: student-information-management-application/src/main/java/com/student/information/management/add/app/model/student/Student.java [hunk 10]
```diff
@@ -0,0 +1,204 @@
...
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * This is the Student Model.
+ */
+public class Student {
```
Comment (RoCS2024) [inline student-information-management-application/src/main/java/com/student/information/management/add/app/model/student/Student.java:11]: keep the POJO clean. remove unnecessary methods

File: student-information-management-application/src/main/java/com/student/information/management/add/app/facade/student/StudentFacade.java [hunk 11]
```diff
@@ -0,0 +1,14 @@
...
+import com.student.information.management.add.app.model.student.Student;
+
+import java.util.List;
+
+/**
+ * This is the Student Facade.
+ */
+public interface StudentFacade {
+    List<Student> getAllStudents();
+    Student getStudentById(String student_id) throws RuntimeException;
```
Comment (RoCS2024) [inline student-information-management-application/src/main/java/com/student/information/management/add/app/facade/student/StudentFacade.java:12]: fix exception

File: student-information-management-application/src/main/java/com/student/information/management/add/app/facade/student/impl/StudentFacadeImpl.java [hunk 12]
```diff
@@ -0,0 +1,36 @@
...
+import com.student.information.management.add.app.model.student.Student;
+import com.student.information.management.add.data.student.dao.StudentDao;
+import com.student.information.management.add.data.student.dao.impl.StudentDaoImpl;
+
+import java.util.List;
+
+/**
+ * This is the StudentFacadeImpl.
+ */
+public class StudentFacadeImpl implements StudentFacade {
```
Comment (RoCS2024) [inline student-information-management-application/src/main/java/com/student/information/management/add/app/facade/student/impl/StudentFacadeImpl.java:13]: put the validations here

File: student-information-management-application/src/main/test/com/student/information/management/add/app/facade/student/impl/StudentFacadeImplTest.java [hunk 13]
```diff
@@ -0,0 +1,18 @@
...
+
+import com.student.information.management.add.app.model.student.Student;
+import org.junit.jupiter.api.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+/**
+ * This is the Student Facade Impl Test.
+ */
+public class StudentFacadeImplTest {
```
Comment (RoCS2024) [inline student-information-management-application/src/main/test/com/student/information/management/add/app/facade/student/impl/StudentFacadeImplTest.java:11]: test other methods from StudentFacadeImpl

File: student-information-management-application/src/main/test/com/student/information/management/add/data/student/dao/impl/StudentDaoImplTest.java [hunk 14]
```diff
@@ -0,0 +1,56 @@
...
+ * This is the Student Dao Impl Test.
+ */
+public class StudentDaoImplTest {
+    @Mock
+    private StudentDao studentDao;
+    private List<Student> students;
+
+    @Before
+    public void setUp() {
+        MockitoAnnotations.openMocks(this);
```
Comment (RoCS2024) [inline student-information-management-application/src/main/test/com/student/information/management/add/data/student/dao/impl/StudentDaoImplTest.java:28]: why is this necessary?

File: student-information-management-application/src/main/test/com/student/information/management/add/data/student/dao/impl/StudentDaoImplTest.java [hunk 15]
```diff
@@ -0,0 +1,56 @@
...
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.when;
+
+/**
+ * This is the Student Dao Impl Test.
+ */
+public class StudentDaoImplTest {
```
Comment (RoCS2024) [inline student-information-management-application/src/main/test/com/student/information/management/add/data/student/dao/impl/StudentDaoImplTest.java:21]: test other methods in StudentDaoImpl

**Review by RoCS2024 — CHANGES_REQUESTED** [review]
see comments. 
also, why do you have lib directory here? what is the purpose of that?

File: student-information-management-application/src/main/java/com/student/information/management/add/data/student/dao/StudentDao.java (hunk 6 above)
Comment (katieann7) [reply student-information-management-application/src/main/java/com/student/information/management/add/data/student/dao/StudentDao.java:12]: All Done po.

File: student-information-management-application/src/main/java/com/student/information/management/add/data/student/dao/impl/StudentDaoImpl.java (hunk 7 above)
Comment (katieann7) [reply student-information-management-application/src/main/java/com/student/information/management/add/data/student/dao/impl/StudentDaoImpl.java:23]: All Done po.

File: student-information-management-application/src/main/java/com/student/information/management/add/data/student/dao/impl/StudentDaoImpl.java (hunk 8 above)
Comment (katieann7) [reply student-information-management-application/src/main/java/com/student/information/management/add/data/student/dao/impl/StudentDaoImpl.java:37]: All Done po.

File: student-information-management-application/src/main/java/com/student/information/management/add/app/model/student/Student.java (hunk 9 above)
Comment (katieann7) [reply student-information-management-application/src/main/java/com/student/information/management/add/app/model/student/Student.java:27]: Deleted na po.

File: student-information-management-application/src/main/java/com/student/information/management/add/app/model/student/Student.java (hunk 10 above)
Comment (katieann7) [reply student-information-management-application/src/main/java/com/student/information/management/add/app/model/student/Student.java:11]: Deleted na po.

File: student-information-management-application/src/main/java/com/student/information/management/add/app/facade/student/StudentFacade.java (hunk 11 above)
Comment (katieann7) [reply student-information-management-application/src/main/java/com/student/information/management/add/app/facade/student/StudentFacade.java:12]: All Done po.

File: student-information-management-application/src/main/test/com/student/information/management/add/app/facade/student/impl/StudentFacadeImplTest.java (hunk 13 above)
Comment (katieann7) [reply student-information-management-application/src/main/test/com/student/information/management/add/app/facade/student/impl/StudentFacadeImplTest.java:11]: All Done po.

File: student-information-management-application/src/main/test/com/student/information/management/add/data/student/dao/impl/StudentDaoImplTest.java (hunk 14 above)
Comment (katieann7) [reply student-information-management-application/src/main/test/com/student/information/management/add/data/student/dao/impl/StudentDaoImplTest.java:28]: It is necessary po for testing the addStudent. If the @Test will pass or not.

**Comment by katieann7** [comment]
> see comments. also, why do you have lib directory here? what is the purpose of that?

Sir it is for test po.

File: student-information-management-application/src/main/test/com/student/information/management/add/data/student/dao/impl/StudentDaoImplTest.java (hunk 14 above)
Comment (katieann7) [reply student-information-management-application/src/main/test/com/student/information/management/add/data/student/dao/impl/StudentDaoImplTest.java:28]: Already Fix po.

**Review by JustineDaveMagnaye — APPROVED** [review]

**Review by RoCS2024 — APPROVED** [review]
